src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,248 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+    private static StorageDao storageDao;
+    private static RegistrationService registrationService;
+    private User firstUser;
+    private User secondUser;
+    private User thirdUser;
+    private User fourthUser;
+    private User fifthUser;
+    private User sixthUser;
+    private User seventhUser;
+    private User eighthUser;
+    private User nineUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @BeforeEach
+    void setUp() {
+        firstUser = new User();
+        firstUser.setId(1L);
+        firstUser.setLogin("tarsan@Gmail.com");
+        firstUser.setPassword("123456");
+        firstUser.setAge(30);
```

Comment:
let's set up only one valid user here, and change some fields in each test (set too short password etc). it will be easier to read, now we have to find what was wrong in `thirdUser` that is defined at the beginning of the class